RecoJets/JetProducers/src/JetSpecific.cc

```diff
@@ -397,6 +397,7 @@ bool reco::makeSpecific(std::vector<reco::CandidatePtr> const& mcparticles, GenJ
         case 13:  // muon
           genJetSpecific->m_MuonEnergy += e;
           ++(genJetSpecific->m_MuonMultiplicity);
+          [[fallthrough]];
```

Comment:
If the muon energy/multiplicity has to be added to the charged hadron energy/multiplicity this is also correct, but please @rappoccio @knash @gkasieczka confirm

Reply:
@rappoccio @knash @gkasieczka did you have time to look at it? I think this is correct, but these warnings could be a good opportunity to have a further quick review of the code instead of limiting ourselves to blindly silence them

Reply:
This looks like a mistake on my part. I think we shouldn't put the muon fraction in with the charged hadron fraction so the "fallthrough" should be a "break".

Reply:
Thank you @rappoccio

If this is the fix that jetmet requests, I think @mrodozov can apply it already in this PR: can you please proceed Mircho? Indeed these "technical" pull requests aiming to fix compiler warnings resulted quite useful in identifying and fixing subtle bugs in the code as this one.

I'd expect some non negligible effect on pf gen jets with muons, let see the comparisons when the PR is merged and the tests are run.

Reply:
Yes, this will non-negligibly affect the pf gen jets (and hopefully will make things work better).